js/entity_browser.new_display.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file entity_browser.entity_reference.js
+ *
+ */
+(function ($, Drupal, drupalSettings) {
+
+  "use strict";
+
+  /**
+   * Registers behaviours related to selected entities.
+   */
+  Drupal.behaviors.entityBrowserNewDisplay = {
+    attach: function (context) {
+      $(context).find('.entity-browser-test-files-form').each(function () {
+        $(this).find('.selected-entities-list').sortable({
+          stop: Drupal.entityBrowserNewDisplay.entitiesReordered
+        });
+      });
+    }
+  };
+
+  Drupal.entityBrowserNewDisplay = {};
+
+  /**
+   * Reacts on sorting of the entities.
+   *
+   * @param event
+   *   Event object.
+   * @param ui
+   *   Object with detailed information about the sort event.
+   */
+  Drupal.entityBrowserNewDisplay.entitiesReordered = function(event, ui) {
+    var items = $(this).find('.selected-item-container');
```

Comment:
We need to remove target_id hidden element and add weights on entities (see below). This function will need to be updated to reflect that change.